common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala

```diff
@@ -3756,6 +3760,24 @@ object CelebornConf extends Logging {
       .checkValues(Set(ShuffleMode.HASH.name, ShuffleMode.SORT.name))
       .createWithDefault(ShuffleMode.HASH.name)
 
+  val CLIENT_PUSH_WRITER_MODE_BY_PARTITION_COUNT_THRESHOLD: ConfigEntry[Int] =
+    buildConf("celeborn.client.spark.push.writer.modeByPartitionCount.threshold")
+      .categories("client")
+      .doc("Threshold of shuffle partition count for push writer mode by partition count.")
+      .version("0.4.0")
+      .intConf
+      .createWithDefault(2000)
+
+  val CLIENT_PUSH_WRITER_MODE_BY_PARTITION_COUNT_ENABLED: ConfigEntry[Boolean] =
+    buildConf("celeborn.client.spark.push.writer.modeByPartitionCount.enabled")
```

Comment:
The suggested configuration names are
```
celeborn.client.spark.push.dynamicWriteMode.enabled
celeborn.client.spark.push.dynamicWriteMode.partitionNum.threshold
```

Usually, we describe the global switch briefly, but detailed in each sub-conditions

e.g.
```
celeborn.client.spark.push.dynamicWriteMode.enabled

Whether to dynamically switch push write mode based on conditions.
```
```
celeborn.client.spark.push.dynamicWriteMode.partitionNum.threshold

... balabala
This configuration only takes effect when XXX is enabled.
```

Also please describe the relationship between the existing configuration `celeborn.client.spark.shuffle.writer`

Reply:
Thanks for your suggestion~.Updated